plugins/jvm-abi-gen/src/org/jetbrains/kotlin/jvm/abi/JvmAbiOutputExtension.kt

```diff
@@ -217,3 +203,33 @@ class JvmAbiOutputExtension(
         }
     }
 }
+
+private class BodyStrippingMethodVisitor(visitor: MethodVisitor?) : MethodVisitor(Opcodes.API_VERSION, visitor) {
+    override fun visitCode() {
+        with(mv) {
+            visitCode()
+            visitInsn(Opcodes.ACONST_NULL)
+            visitInsn(Opcodes.ATHROW)
```

Comment:
I think we can throw something more meaningful than `throw null` → `Exception in thread "main" java.lang.NullPointerException`

Maybe something like `NullPointerException("Kotlin ABI jar got linked into runtime binary")` or something to help people with wrongly configured build that linked ABI jar instead of a full jar.

Reply:
I think it makes sense, although I consider this case to be extremely rare so I am not sure how much value it adds. Also I am not sure if it affect the size of the ABI significantly and if yes - how much. 

It is out of the scope of this change, @udalov If you think it is worth it - let me know, I will open a separate PR.